pkg/labels/v2/cache.go

```diff
@@ -25,7 +29,8 @@ func newCache[T comparable]() *cache[T] {
 // cache is a sync.Pool-based cache of recently created Label/Labels
 // to reduce allocations.
 type cache[T comparable] struct {
-	pool sync.Pool
+	pool      sync.Pool
+	hit, miss atomic.Int32 // TODO remove? costs about 4ns or ~10%
```

Comment:
How expensive is a `if option.Debug { ... }` when Debug is false? I could see it going either way.